pkg/utils/utils_linux.go

```diff
@@ -91,9 +100,26 @@ func determineEndian() binary.ByteOrder {
 	return endian
 }
 
-// Get all the veth interfaces.
-// Similar to ip link show type veth
-func GetInterface(name string, interfaceType string) (netlink.Link, error) {
+// GetOutgoingInterface gets the outgoing interface by first trying
+// executing an equivalent to `ip route get <External IP>`
+// and fall back to `ip link show type veth`
+func GetOutgoingInterface(fallbackInterfaceName string, interfaceType string) (netlink.Link, error) {
```

Comment:
We want to change this behavior to use the device associated with the default route instead of just checking `eth0`. Equivalent of `ip route show default type device`.